photon-api/src/main/scala/com/linkedin/photon/ml/supervised/classification/SmoothedHingeLossLinearSVMModel.scala

```diff
@@ -33,35 +34,91 @@ class SmoothedHingeLossLinearSVMModel(override val coefficients: Coefficients)
   with Serializable {
 
   /**
-   * Compute the mean of the soft hinge loss linear SVM model
+   *
+   * @return The model type.
+   */
+  override def modelType: TaskType = SMOOTHED_HINGE_LOSS_LINEAR_SVM
```

Comment:
We're going to have to do this currently somewhere, either within the class itself or in some util method which maps `LogisticRegressionModel` to 'TaskType.LOGISTIC_REGRESSION`, etc.

If you've got a suggestion for another way, I could replace it, though.